File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private static final int CARS_PER_HOUR = 221;

    public double productionRatePerHour(int speed) {
        double carsProduced = speed * CARS_PER_HOUR;

        if (speed == 10) {
            return 0.77 * carsProduced;
        } else if (speed == 9) {
            return 0.80 * carsProduced;
        } else if ( speed >= 5 && speed <= 8) {
            return 0.90 * carsProduced;
        } else {
            return carsProduced;
        }
    }

    public int workingItemsPerMinute(int speed) {
        double ratePerHour = productionRatePerHour(speed);

        return (int)ratePerHour / 60;
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    private const int CarsPerHour = 221;
    
    public static double SuccessRate(int speed) => speed switch
    {
        10 => 0.77,
        9 => 0.80,
        <= 8 and >= 5 => 0.90,
        <= 4 and >= 1 => 1.00,
        _ => 0.00
    };
    
    public static double ProductionRatePerHour(int speed) => CarsPerHour * speed * SuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed) => (int)ProductionRatePerHour(speed) / 60;
}
